Common/GPU/Vulkan/VulkanQueueRunner.cpp

```diff
@@ -1526,21 +1535,68 @@ void VulkanQueueRunner::PerformRenderPass(const VKRStep &step, VkCommandBuffer c
 
 		case VKRRenderCommand::DRAW_INDEXED:
 		{
+			static const char indexShift[2] = { 1, 2 };
+
 			vkCmdBindDescriptorSets(cmd, VK_PIPELINE_BIND_POINT_GRAPHICS, pipelineLayout, 0, 1, &c.drawIndexed.ds, c.drawIndexed.numUboOffsets, c.drawIndexed.uboOffsets);
-			vkCmdBindIndexBuffer(cmd, c.drawIndexed.ibuffer, c.drawIndexed.ioffset, (VkIndexType)c.drawIndexed.indexType);
-			VkDeviceSize voffset = c.drawIndexed.voffset;
-			vkCmdBindVertexBuffers(cmd, 0, 1, &c.drawIndexed.vbuffer, &voffset);
-			vkCmdDrawIndexed(cmd, c.drawIndexed.count, c.drawIndexed.instances, 0, 0, 0);
+			int indexOffset = 0;
+
+			// Conservative limits for bind eliminination for Adreno. Still a good effect.
+			if (lastIndexOffset + c.drawIndexed.count > 32768) {
+				lastIndexBuffer = 0;
+			}
+
+			if (c.drawIndexed.ibuffer == lastIndexBuffer && c.drawIndexed.indexType == lastIndexType) {
+				indexOffset = (c.drawIndexed.ioffset - lastIndexOffset) >> (c.drawIndexed.indexType + 1);  // NOTE: This only works for UINT16 = 0 and UINT32 = 1.
```

Comment:
nit: Probably meant to use `indexShift`?

-[Unknown]

Reply:
Oh I realized the lookup table `indexShifts` was unnecessary and could just be replaced with this simple expression, since we only support 16 and 32-bit indices here.